libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -715,6 +709,39 @@ void AP_Logger_File::stop_logging(void)
     }
 }
 
+/*
+  wrapper for PrepForArming to move start_new_log to the logger thread
+ */
+void AP_Logger_File::PrepForArming()
+{
+#if APM_BUILD_TYPE(APM_BUILD_Replay)
+    AP_Logger_Backend::PrepForArming();
+#else
+    if (_rotate_pending) {
+        _rotate_pending = false;
+        stop_logging();
+    }
+    if (logging_started()) {
+        return;
+    }
+    uint32_t start_ms = AP_HAL::millis();
+    const uint32_t open_limit_ms = 250;
+
+    /*
+      log open happens in the io_timer thread. We allow for a maximum
+      of 250ms to complete the open
+     */
+    start_new_log_pending = true;
+    EXPECT_DELAY_MS(250);
```

Comment:
I have a sneaking suspicion we might need to increase this.

Reply:
i hope we don't ....